### ISSUE TITLE
View multiple subreddits in tabs

### ISSUE THREAD

**Issue by spcmd** [michael-lazar/rtv#222]
It would be great if we could have tabs for keeping different subreddits loaded/opened all the time and quickly switch between them.

Futhermore, rtv could run as a "single instance" and it could load the reddit links/urls to separate tabs. This way we could handle reddit links coming from external applications and also the **rtv -l <reddit link>** command could open the link in a new tab (in the already running rtv instance).

[**Example image**](http://i.imgur.com/nmEkgrZ.png) (see the top bar, it could be the tab list) (optional: every tab could be numbered automatically, like in some Vim-like browsers).

**Comment by michael-lazar** [comment]
Hi, thanks for the feedback!

> It would be great if we could have tabs for keeping different subreddits loaded/opened all the time and quickly switch between them.

If you're doing that, why not just use tmux to manage the windows/tabs? On the other hand, I could see adding something like a subscription toolbar to make it easier to select from your favorite subreddits.

> Futhermore, rtv could run as a "single instance" and it could load the reddit links/urls to separate tabs. This way we could handle reddit links coming from external applications and also the rtv -l command could open the link in a new tab (in the already running rtv instance).

I'm not sure I understand what you mean by a "single instance". Do you mean like a daemon? It sounds like your proposal would mean that you wouldn't be able to open two instances of rtv at once.

**Comment by spcmd** [comment]
Yes, tmux could be used, but not everyone uses tmux, I just thought a "native" tab handling would a be better solution. And despite of using tmux, every rtv instance is running in a separate process using "extra" resources. Maybe a tabbed mode could use a single process but I guess that would mean a "daemon" mode.

For the "single instance": I mean, like a web browser. It's running, using tabs, and for example when I click on a link in an external application it gets opened in the already running browser, in a tab, instead of opening a new window. So for example when I launch this command: **rtv -l redditlink** it would be opened in a tab in the already running rtv instance.

If this would be difficult to implement and can't be done without drastic code changes then feel free to close this issue, I just had this idea and I thought I would share it. :)

> On the other hand, I could see adding something like a subscription toolbar to make it easier to select from your favorite subreddits.

I think that would be nice.
